pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1432,8 +1434,8 @@ func (c *PodAffinityChecker) satisfiesPodsAffinityAntiAffinity(pod *v1.Pod, node
 	return nil, nil
 }
 
-// PodToleratesNodeTaints checks if a pod tolerations can tolerate the node taints
-func PodToleratesNodeTaints(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+// CheckNodeUnschedulablePredicate checks if a pod can be scheduled on a node with Unschedulable spec.
+func CheckNodeUnschedulablePredicate(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
```

Comment:
nit: Shouldn't the function be CheckNodeSchedulable instead as we return true if it is schedulable? Also, do you want to add a e2e test?

Reply:
prefer to current name, as `Unschedulable` is a spec of node; and add e2e case when TaintNodesByCondition to beta :)